mobify/stencil-icon
# (Old) Initial implementation

This is an initial implementation of the Icon component. Right now it only supports SVG. I think we also want to support icon fonts, but that makes things much harder. For now I just want to get initial feedback on the API and usage. Basically, this component renders an icon from an SVG sprite. For details, see the Details section.

Status: **Ready for Initial Review**
Owner: @ry5n
Reviewers: @jeffkamo @avelinet @yourpalsonja and everyone
## Details
- Declares a default icon set ([shoppicon](http://github.com/ry5n/shoppicon)) as a Bower dependency. However, as a developer you need to add the sprite to the page yourself. This is a little disappointing given Stencil’s goal of “it just works” but I think it’s a pragmatic decision for the time being. We’ll need to have a how-to in the docs.
- The dust file assumes your svg symbol ids are formatted like so: `id="{prefix}-{name}"`, and requires you to pass the prefix as a param if it isn’t `icon`. This could be awkward.
- You can declare an icon as scriptable with the `dynamic="true"` param. The main use case here is to change the icon in a view script without having to muck with the component’s internals – instead, you use the public interface in the form of the `name` method: `$('#my-icon').data('component').name('something')`. The fact that you need to opt-in to the UI script should help performance.
- The `title` param allows you to make the icon available to screen readers.
## Jira Tickets:
- [STEN-13]
## Todos:
- [ ] UI Dev +1
## How to Test
- clone this repo
- `cd stencil-icon`
- `npm install && bower install && bundle install`
- run `grunt serve`
- Check the command line output to determine which port the server started at (defaults to 3000)
- Navigate to `localhost:{port}/tests/visual/`
